graylog2-server/src/main/java/org/graylog/plugins/views/search/searchtypes/pivot/MayHaveUnit.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.views.search.searchtypes.pivot;
+
+import org.graylog.plugins.formatting.units.model.UnitId;
+
+public interface MayHaveUnit {
+
+    UnitId unit();
```

Comment:
We should use an `Optional`, as it is declared `Nullable` in all usages:
```suggestion
    Optional<UnitId> unit();
```

Reply:
Just to be sure... 
There are many people in Java community that discourage the use of optionals in fields.
This change will result in having optional fields in our auto value classes.
Are we ok with that? 
Do we stand on this side of the barricade that believes a usage of optionals as fields is a good practice?

Reply:
The only real reason against using `Optional`'s for fields is that it is not `Serializable`. This is not a problem for us, as we do not rely on it and Jackson can serialize it properly (we are using it in lots of DTOs already).